pkg/collector/corechecks/sbom/processor.go

```diff
@@ -259,6 +262,70 @@ func (p *processor) triggerHostScan() {
 	}
 }
 
+func (p *processor) triggerProcfsScan(ctr *workloadmeta.Container) {
+	// Allowed only on Fargate instance for now
+	if !fargate.IsFargateInstance() {
+		log.Warnf("procfs scan supported only on Fargate instances, ignoring : %s", ctr.ID)
+		return
+	}
+
+	if !pkgconfigsetup.Datadog().GetBool("sbom.container.enabled") {
```

Comment:
the `sbom.host.enabled` config is passed when creating the processor. It would be great to do the same thing here https://github.com/DataDog/datadog-agent/blob/d5a84b2612bdfaa746f5f8b0ed99d27806d5c1fe/pkg/collector/corechecks/sbom/check.go#L163

Reply:
and IMO the fargate check could be done in the same spot, when creating the processor. Otherwise we are going to do the same check for each container event

Reply:
I can pass the boolean in the processor creation but I'm not sure there is a huge value, we'll still have to check on each container events as this is done for host.

Reply:
I changed the logic.